src/dataflow/src/metrics/mod.rs

```diff
@@ -53,7 +53,8 @@ impl Metrics {
         };
         let success_label = if success { "success" } else { "error" };
         self.events_read
-            .with_label_values(&[format_label, success_label])
+            .get_delete_on_drop_counter(vec![format_label.to_string(), success_label.to_string()])
+            .leak()
```

Comment:
🤔 so, ... we're creating a delete-on-drop metric, then never drop it? I think I understand why you're doing it, but it seems a bit extra

Reply:
Thinking more, I feel this might be dangerous if we create multiple of these d-o-d wrappers for the same label - you have to be really careful to always call leak on them. Is there a method missing that we could put on these vec's to make it more convenient?

Reply:
I removed the `leak` function and instead expose the underlying metrics within a `raw` module as part of the metrics. This avoids bad API design and still gives clients a way to control the lifetime of metrics independent of Rust's drop semantics.